### PR TITLE
Update Bevy Book for 0.11

### LINKED ISSUE #708: bevy book needs to be migrated to 0.11
At the very least it isn't using the new `.add_systems` api.

https://bevyengine.org/learn/book

### PR DESCRIPTION
**Pull request by cart**
Resolves #708

### CODE REVIEW

File: content/learn/book/getting-started/resources/_index.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ fn greet_people(
     // if that caused the timer to finish, we say hello to everyone
     if timer.0.tick(time.delta()).just_finished() {
         for name in &query {
-            println!("hello {}!", name.0);
+            println!("hello {name}!");
```
Comment (Selene-Amanita) [inline content/learn/book/getting-started/resources/_index.md:48]:
```suggestion
            println!("hello {name.0}!");
```
`Name` doesn't implement display so this wouldn't work, I think.

**Review by Selene-Amanita — CHANGES_REQUESTED** [review]

File: content/learn/book/getting-started/resources/_index.md (hunk 1 above)
Comment (cart) [reply content/learn/book/getting-started/resources/_index.md:48]: Name does implement Display!

Comment (Selene-Amanita) [reply content/learn/book/getting-started/resources/_index.md:48]: @cart Are we talking about this `Name`: https://docs.rs/bevy/latest/bevy/core/struct.Name.html

Or the `Name` defined here: https://bevyengine.org/learn/book/getting-started/ecs/

In the context of the book, I think it's the later, and I don't see Display implemented on it?

Comment (cart) [reply content/learn/book/getting-started/resources/_index.md:48]: Ahhhh yeah you're right. I was validating code by pasting it into something that had `bevy::prelude::*`. Nice catch /  thanks for doubling down!

**Review by Selene-Amanita — APPROVED** [review]

**Pull request merged**